nx/source/services/clkrst.c

```diff
@@ -1,156 +1,53 @@
-#include "types.h"
-#include "result.h"
-#include "arm/atomics.h"
-#include "kernel/ipc.h"
+#define NX_SERVICE_ASSUME_NON_DOMAIN
+#include "service_guard.h"
 #include "services/pcv.h"
-#include "runtime/hosversion.h"
-#include "services/sm.h"
 #include "services/clkrst.h"
+#include "runtime/hosversion.h"
 
 static Service g_clkrstSrv;
-static u64 g_refCnt;
-
-Result clkrstInitialize(void) {
-    if(hosversionBefore(8,0,0)) {
-        return MAKERESULT(Module_Libnx, LibnxError_IncompatSysVer);
-    }
-
-    atomicIncrement64(&g_refCnt);
 
-    if (serviceIsActive(&g_clkrstSrv)) {
-        return 0;
-    }
+NX_GENERATE_SERVICE_GUARD(clkrst);
 
-    Result rc = smGetService(&g_clkrstSrv, "clkrst");
-
-    if (R_FAILED(rc)) {
-        clkrstExit();
-    }
-
-    return rc;
+Result _clkrstInitialize(void) {
+    return smGetService(&g_clkrstSrv, "clkrst");
```

Comment:
The original had a a hosver check, add it here. Also don't need 8.0+ hosver checks in the rest of clkrst since it would be checked here.